earthfile2llb/state_wait_item.go

```diff
@@ -0,0 +1,26 @@
+package earthfile2llb
+
+import (
+	"github.com/earthly/earthly/util/llbutil/pllb"
+	"github.com/earthly/earthly/util/waitutil"
+)
+
+type stateWaitItem struct {
+	c     *Converter
+	state *pllb.State
+}
+
+// SetDoPush has no effect, but exists to satisfy interface
+func (swi *stateWaitItem) SetDoPush() {
+}
+
+// SetDoSave has no effect, but exists to satisfy interface
+func (swi *stateWaitItem) SetDoSave() {
+}
+
+func newStateWaitItem(state *pllb.State, c *Converter) waitutil.WaitItem {
+	return &stateWaitItem{
+		c:     c,
+		state: state,
+	}
+}
```

Comment:
I would move the WaitItems to their own sub-package, to keep things a little more organized.

Reply:
can we move all the `wait_block.go` code into `earthfile2llb/waitblock`? it accesses a lot of converter internals.

or are you thinking simply these saveImageWaitItem/saveArtifactWaitItem/saveStateItem structs?

Reply:
I was thinking just these wait items. If it's even possible anyway.

Reply:
```
            +earthly | 	imports github.com/earthly/earthly/earthfile2llb/waititem
            +earthly | 	imports github.com/earthly/earthly/earthfile2llb: import cycle not allowed
```

unfortunately it's not going to work, since `waititem.SaveImageWaitItem` need to reference `earthfile2llb.Converter`, and the converter code needs to access `waititem.NewSaveImageWaitItem`.

Reply:
Yeah, ok makes sense.